che-incubator/che-code
# Launching vscode with some default extensions installed

### What does this PR do?

Allows to start the editor with some default extensions.

The idea is to upload the extension binary to the container (using a post start command for instance) and then define `DEFAULT_EXTENSIONS` environment variable to allow the editor use the extension on the startup.

### What issues does this PR fix?

https://issues.redhat.com/browse/CRW-7246

### How to test this PR?

- create a workspace using the editor image [quay.io/che-incubator-pull-requests/che-code:pr-424-amd64](https://quay.io/che-incubator-pull-requests/che-code:pr-424-amd64)

once workspace started, ensure no extensions are installed (except built-in)

- open the devfile, add post start command to download the extensions and then restart the workspace from local devfile

```
  - id: add-default-extensions
    exec:
      # put your tooling container name here
      component: runtime
      commandLine: |
        # download regular binary
        curl https://open-vsx.org/api/atlassian/atlascode/3.0.10/file/atlassian.atlascode-3.0.10.vsix --location -o /tmp/atlassian.atlascode-3.0.10.vsix
        curl https://open-vsx.org/api/snowflake/snowflake-vsc/1.9.1/file/snowflake.snowflake-vsc-1.9.1.vsix --location -o /tmp/snowflake.snowflake-vsc-1.9.1.vsix

events:
  postStart:
    - add-default-extensions
```

- add the configmap, the workspace should be restarted
```
kind: ConfigMap
apiVersion: v1
metadata:
  name: default-extensions
  labels:
    controller.devfile.io/mount-to-devworkspace: 'true'
    controller.devfile.io/watch-configmap: 'true'
  annotations:
    controller.devfile.io/mount-as: env
data:
  DEFAULT_EXTENSIONS: '/tmp/atlassian.atlascode-3.0.10.vsix;/tmp/snowflake.snowflake-vsc-1.9.1.vsix'
```

- refresh the editor page, both extensions should be installed.

### Does this PR contain changes that override default upstream Code-OSS behavior?
